src/System.ServiceModel.Tests.Common/src/OuterLoopTests.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Newtonsoft.Json;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+
+namespace System.ServiceModel.Tests.Common
+{
+    public abstract class OuterLoopTests : IDisposable
+    {
+        protected string EndpointAddress
+        {
+            get;
+            private set;
+        }
+
+        private string BaseAddress
+        {
+            // Would like to pull this address either from msbuild properties, env vars, or 
+            // configuration passed into xunit.
+            get { return "http://localhost:44283"; }
+        }
+
+        public OuterLoopTests(string resourceName)
+        {
+            MakeConfigRequest();
+            EndpointAddress = MakeResourceRequest(resourceName);
+        }
+
+        public void Dispose()
+        {
+            // Placeholder for releasing the resource
+        }
+
+        private void MakeConfigRequest()
+        {
+            using (HttpClient httpClient = new HttpClient())
+            {
+                httpClient.BaseAddress = new Uri(BaseAddress);
+                var content = new StringContent(
+                    @"{ resourcesDirectory : ""C:\\dotnet\\wcf\\src\\System.Private.ServiceModel\\tests\\Scenarios\\SelfHostWcfService\\bin\\Debug"" }",
```

Comment:
@roncain this along with the address for bridge should be passed in here somehow ..I guess you already have a work item for this.